pkg/gardenlet/operation/botanist/dualstackmigration_test.go

```diff
@@ -0,0 +1,339 @@
+// SPDX-FileCopyrightText: 2025 SAP SE or an SAP affiliate company and Gardener contributors
```

Comment:
Please avoid using mocks for new tests, see https://github.com/gardener/gardener/blob/master/docs/development/testing.md

> Generally, prefer fakes over mocks, e.g., use controller-runtime fake client over mock clients.

Reply:
I refactored the code so that I extracted the actual logic to its own function, to avoid mocks and fakes.